### PR TITLE
Update cli and client

### PR DESCRIPTION
**Pull request by MikailBag**

### CODE REVIEW

File: src/cli/src/login.rs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    .unwrap()
+}
+
+pub(crate) async fn exec(opt: &Opt) -> anyhow::Result<()> {
+    let endpoint = match &opt.endpoint {
+        Some(ep) => ep.clone(),
+        None => ask("JJS apiserver endpoint", Some("http://localhost:1779/")).await?,
+    };
+    let username = ask("Username", None).await?;
+    let password = tokio::task::spawn_blocking(|| -> anyhow::Result<String> {
```
Comment (pavel-the-best) [inline src/cli/src/login.rs:35]: Remove duplicated code

File: src/cli/src/main.rs [hunk 2]
```diff
@@ -1,42 +1,42 @@
...
 
-use client::ApiClient;
+use clap::Clap;
 use std::process::exit;
-use structopt::StructOpt;
 
 /// Command-line client for JJS
 ///
 /// To get Bash completion, run:
 /// COMPLETION=1 <path/to/jjs-cli> > /tmp/compl.sh
```
Comment (pavel-the-best) [inline src/cli/src/main.rs:15]: argv[0]?

File: src/cli/src/toolchains.rs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    pull: bool,
+}
+
+pub(crate) async fn exec(opt: &Opt, api: &client::ApiClient) -> anyhow::Result<()> {
+    if opt.pull {
+        println!("pulling the image");
+        let mut cmd = tokio::process::Command::new("docker");
+        cmd.arg("pull");
+        cmd.arg(&opt.image);
+        let status = cmd.status().await.context("docker not available")?;
```
Comment (pavel-the-best) [inline src/cli/src/toolchains.rs:23]: Why is docker required?

**Comment by MikailBag** [comment]
bors r=pavel-the-best

**Pull request merged**